src/solidity-frontend/solidity_convert.cpp

```diff
@@ -0,0 +1,1898 @@
+
+  SolidityGrammar::StatementT type = SolidityGrammar::get_statement_t(stmt);
+  msg.status(fmt::format(
+    "	@@@ got Stmt: SolidityGrammar::StatementT::{}",
+    SolidityGrammar::statement_to_str(type)));
+
+  switch(type)
+  {
+  case SolidityGrammar::StatementT::Block:
+  {
+    get_block(stmt, new_expr);
+    break;
+  }
+  case SolidityGrammar::StatementT::ExpressionStatement:
+  {
+    get_expr(stmt["expression"], new_expr);
+    break;
+  }
+  case SolidityGrammar::StatementT::VariableDeclStatement:
+  {
+    const nlohmann::json &declgroup = stmt["declarations"];
+
+    codet decls("decl-block");
+    unsigned ctr = 0;
+    // N.B. Although Solidity AST JSON uses "declarations": [],
+    // there will ALWAYS be 1 declaration!
+    // A second declaration will become another stmt in "statements"
+    for(const auto &it : declgroup.items())
+    {
+      // deal with local var decl with init value
+      nlohmann::json decl = it.value();
+      if(stmt.contains("initialValue"))
+      {
+        // Need to combine init value with the decl JSON object
+        decl["initialValue"] = stmt["initialValue"];
+      }
+
+      exprt single_decl;
+      if(get_var_decl_stmt(decl, single_decl))
+        return true;
+
+      decls.operands().push_back(single_decl);
+      ++ctr;
+    }
+    msg.status(fmt::format(" \t@@@ DeclStmt group has {} decls", ctr));
+
+    new_expr = decls;
+    break;
+  }
+  case SolidityGrammar::StatementT::ReturnStatement:
+  {
+    if(!current_functionDecl)
+      assert(!"Error: ESBMC could not find the parent scope for this ReturnStatement");
+
+    // 1. get return type
+    // TODO: Fix me! Assumptions:
+    //  a). It's "return <expr>;" not "return;"
+    //  b). <expr> is pointing to a DeclRefExpr, we need to wrap it in an ImplicitCastExpr as a subexpr
+    assert(stmt.contains("expression"));
+    assert(stmt["expression"].contains("referencedDeclaration"));
+
+    typet return_type;
+    if(get_type_description(
+         stmt["expression"]["typeDescriptions"], return_type))
+      return true;
+
+    // 2. get return value
+    code_returnt ret_expr;
+    const nlohmann::json &rtn_expr = stmt["expression"];
+    // wrap it in an ImplicitCastExpr to convert LValue to RValue
+    nlohmann::json implicit_cast_expr =
+      make_implicit_cast_expr(rtn_expr, "LValueToRValue");
+
+    exprt val;
+    if(get_expr(implicit_cast_expr, val))
+      return true;
+
+    solidity_gen_typecast(ns, val, return_type);
+    ret_expr.return_value() = val;
+
+    new_expr = ret_expr;
+    break;
+  }
+  case SolidityGrammar::StatementT::ForStatement:
+  {
+    // Based on rule for-statement
+    // TODO: Fix me. Assuming for loop contains everything
+    assert(
+      stmt.contains("initializationExpression") && stmt.contains("condition") &&
+      stmt.contains("loopExpression") && stmt.contains("body"));
+    // TODO: Fix me. Currently we don't support nested for loop
```

Comment:
Remember to open an issue for this

Reply:
ACK. 
will do it when extending the new frontend to support more benchmarks.